private/Producers/FbxProducer/FbxProducerImpl.cpp

```diff
@@ -954,64 +958,126 @@ cd::BoneID FbxProducerImpl::AddBone(const fbxsdk::FbxNode* pSDKNode, cd::BoneID
 
-		animation.SetDuration(static_cast<float>(end > start ? end - start : 1.0f));
-		animation.SetTicksPerSecond(static_cast<float>(24.0f));
+		std::vector<float> times;
+		times.resize(trackCount);
 
-		float period = 1.f / 24.f; // todo: it can make variable, like 24fps or 60fps...
-		const auto& bones = pSceneDatabase->GetBones();
-		int trackCount = static_cast<int>(std::ceil((end - start) / period));
-		for (uint32_t boneIndex = 0U; boneIndex < pSceneDatabase->GetBoneCount(); ++boneIndex)
+		std::vector<std::vector<cd::Matrix4x4>> worldMatrices;
+		worldMatrices.resize(pSceneDatabase->GetBoneCount());
+		for (int i = 0; i < pSceneDatabase->GetBoneCount(); i++)
 		{
-			fbxsdk::FbxNode* bone = scene->FindNodeByName(bones[boneIndex].GetName());
-			cd::TrackID::ValueType trackHash = cd::StringHash<cd::TrackID::ValueType>(bones[boneIndex].GetName());
-			cd::TrackID trackID = m_trackIDGenerator.AllocateID(trackHash);
+			worldMatrices[i].resize(trackCount);
+		}
+
+		animation.SetDuration(end - start);
+		animation.SetTicksPerSecond(static_cast<float>(30.0f));
+
+		// Goes through the whole timeline to compute animated word matrices.
+		// Fbx sdk seems to compute nodes transformation for the whole scene, so it's
+		// much faster to query all nodes at once for the same time t.
+		FbxAnimEvaluator* evaluator = scene->GetAnimationEvaluator();
+		for (int trackIndex = 0; trackIndex < trackCount; ++trackIndex)
+		{
+			const float time = std::min(trackIndex * period, end - start);
+			times[trackIndex] = time;
+			for (uint32_t boneIndex = 0U; boneIndex < pSceneDatabase->GetBoneCount(); ++boneIndex)
+			{
+				fbxsdk::FbxNode* bone = bones[boneIndex];
+				if (bone)
+				{
+					const FbxAMatrix fbxMatrix = evaluator->GetNodeGlobalTransform(bone, FbxTimeSeconds(time + start));
+					const cd::Matrix4x4 matrix = details::ConvertFbxMatrixToCDMatrix(fbxMatrix);
+					worldMatrices[boneIndex][trackIndex] = matrix;
+				}
 
-			cd::Track boneTrack(trackID, bones[boneIndex].GetName());
-			const char* name = bones[boneIndex].GetName();
+			}
+		}
+
+		// Builds world inverse matrices.
+		std::vector<std::vector<cd::Matrix4x4>> worldInverseMatrices;
+		worldInverseMatrices.resize(pSceneDatabase->GetBoneCount());
+		for (int i = 0; i < pSceneDatabase->GetBoneCount(); i++)
+		{
+		
+			const std::vector<cd::Matrix4x4>& boneWorldMatrices = worldMatrices[i];
+			std::vector<cd::Matrix4x4>& boneWorldInvMatrices = worldInverseMatrices[i];
+			boneWorldInvMatrices.resize((trackCount));
+			for (size_t p = 0; p < trackCount; ++p)
+			{
+				boneWorldInvMatrices[p] = boneWorldMatrices[p].Inverse();
+			}
+		}
+		// Builds local space animation tracks.
+		// Allocates all tracks with the same number of joints as the skeleton.
+		for (int i = 0; i < pSceneDatabase->GetBoneCount(); i++)
+		{
+			std::string str(animStackName);
+			cd::TrackID::ValueType trackHash = cd::StringHash<cd::TrackID::ValueType>(str + bones[i]->GetName());
```

Comment:
```suggestion
			auto trackHash = cd::StringHash<cd::TrackID::ValueType>(str + bones[i]->GetName());
```
From template parameter, we already know cd::TrackID::ValueType here. So use auto is OK.

Reply:
And in this case, use std::format is a better choise to improve performance.